### PR TITLE
CO-1757 Init

### PR DESCRIPTION
**Pull request by jpb**
Usage:

```
    steps:
    - uses: unbounce/actions-deploy@co-1757-init
      with:
        type: ui
```

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -26,6 +26,8 @@ jobs:
     - run: npm run lint:format
     - run: npm run lint
     - run: npm run test:ci
+    - run: npm run build
+    - run: git status -s | grep dist/index.js && echo '::warning Bundle file (dist/index.js) may not be up-to-date' || true
```
Comment (tuff) [inline .github/workflows/push.yml:30]: 👍

File: package.json [hunk 2]
```diff
@@ -36,5 +36,13 @@
...
       "pre-commit": "lint-staged"
     }
+  },
+  "lint-staged": {
+    "*.{ts,tsx}": [
+      "tslint --config tslint.json --project tsconfig.json --fix",
+      "prettier --write",
+      "bash -c 'npm run build && git add dist/index.js'"
+    ],
+    "package(-lock)?.json": ["bash -c 'npm run build && git add dist/index.js'"]
```
Comment (tuff) [inline package.json:46]: Why is `bash -c` needed here?

File: README.md [hunk 3]
```diff
@@ -1 +1,31 @@
...
+    steps:
+    - uses: actions/checkout@master
+    - uses: unbounce/actions-deploy@master
+      with:
+        type: ui # or 'lambda' or 'kube'
+        # Optional parameters with defaults:
+        # master-branch: master
+        # production-environment: production
+        # pre-production-environment: integration
+```
```
Comment (tuff) [inline README.md:31]: Why are the self-hosted job and `if` expressions not included in this snippet?

File: package.json (hunk 2 above)
Comment (jpb) [reply package.json:46]: It is used to swallow up the arguments – lint-staged commands are passed the filename that has changed which causes `npm run build` to error. The alternative is to use [this style](https://github.com/okonet/lint-staged#example-run-tsc-on-changes-to-typescript-files-but-do-not-pass-any-filename-arguments), but the prettier and tslint command should get passed the filenames as parameters, and I couldn't get it working with two different rules).

File: README.md [hunk 4]
```diff
@@ -1 +1,31 @@
 # actions-deploy
+
+GitHub Actions-based Release and Deployment.
+
+## Usage
```
Comment (tuff) [inline README.md:5]: We should add an instruction here like

> Add `.github/workflows/deployment.yml` to your project:

File: README.md (hunk 3 above)
Comment (jpb) [reply README.md:31]: I wasn't sure about this – maybe I'll replace this with an example file rather than maintaining that in the README.

**Pull request merged**